torch/csrc/distributed/c10d/symm_mem/NVSHMEMSymmetricMemory.cu

```diff
@@ -229,6 +232,23 @@ class NVSHMEMSymmetricMemory : public SymmetricMemory {
         .make_tensor();
   }
 
+  at::Tensor get_remote_tensor(
+      int peer,
+      c10::IntArrayRef sizes,
+      c10::ScalarType dtype) override {
+    auto peer_ptr = pai_->buffers_[peer];
+    TORCH_CHECK(peer_ptr != nullptr,
+        "Cannot get peer tensor across nodes, my rank: ", get_rank(), ", peer: ", peer);
+    // Peer tensor's offset is the same as my offset
+    auto data_ptr = reinterpret_cast<uint8_t*>(peer_ptr) + offset_;
```

Comment:
you are not checking that you have enough space in the buffer, like get_buffer does. Would make sense to have one function that checks this, into which both get_buffer and get_remote_tensor will call

Reply:
Never mind. Perhaps the local tensor will experience some `view` ops which will changes its shape. In this case, the user may want to provide the new shape to get a corresponding tensor directly.